pkg/controller/disruption/disruption.go

```diff
@@ -0,0 +1,486 @@
+	dc := &DisruptionController{
+		kubeClient:    kubeClient,
+		podController: podInformer.GetController(),
+		queue:         workqueue.New(),
+	}
+
+	dc.getUpdater = func() updater { return dc.writePdbStatus }
+
+	dc.podLister.Indexer = podInformer.GetIndexer()
+
+	podInformer.AddEventHandler(framework.ResourceEventHandlerFuncs{
+		AddFunc:    dc.addPod,
+		UpdateFunc: dc.updatePod,
+		DeleteFunc: dc.deletePod,
+	})
+
+	dc.pdbStore, dc.pdbController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return dc.kubeClient.Policy().PodDisruptionBudgets(api.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return dc.kubeClient.Policy().PodDisruptionBudgets(api.NamespaceAll).Watch(options)
+			},
+		},
+		&policy.PodDisruptionBudget{},
+		30*time.Second,
+		framework.ResourceEventHandlerFuncs{
+			AddFunc:    dc.addDb,
+			UpdateFunc: dc.updateDb,
+			DeleteFunc: dc.removeDb,
+		},
+	)
+	dc.pdbLister.Store = dc.pdbStore
+
+	dc.rcIndexer, dc.rcController = framework.NewIndexerInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return dc.kubeClient.ReplicationControllers(api.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return dc.kubeClient.ReplicationControllers(api.NamespaceAll).Watch(options)
+			},
+		},
+		&api.ReplicationController{},
+		30*time.Second,
+		framework.ResourceEventHandlerFuncs{},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+
+	dc.rcLister.Indexer = dc.rcIndexer
+
+	dc.rsStore, dc.rsController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return dc.kubeClient.Extensions().ReplicaSets(api.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return dc.kubeClient.Extensions().ReplicaSets(api.NamespaceAll).Watch(options)
+			},
+		},
+		&extensions.ReplicaSet{},
+		30*time.Second,
+		framework.ResourceEventHandlerFuncs{},
+	)
+
+	dc.rsLister.Store = dc.rsStore
+
+	dc.dStore, dc.dController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return dc.kubeClient.Extensions().Deployments(api.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return dc.kubeClient.Extensions().Deployments(api.NamespaceAll).Watch(options)
+			},
+		},
+		&extensions.Deployment{},
+		30*time.Second,
+		framework.ResourceEventHandlerFuncs{},
+	)
+
+	dc.dLister.Store = dc.dStore
+
+	return dc
+}
+
+func (dc *DisruptionController) finders() []podControllerFinder {
+	return []podControllerFinder{dc.getPodReplicationControllers, dc.getPodDeployments}
```

Comment:
IIUC you're covering the case of a ReplicationController and a ReplicaSet that is owned by a Deployment, but not ReplicaSet that was created directly by a user. Is that correct If so, I think you also need to cover that third case, i.e. add ReplicaSets here and have a corresponding function for it.